### PR TITLE
Interpolate between shades of green

### PR DESCRIPTION
**Pull request by birneamstiel**

### CODE REVIEW

File: src/system_simulation/trace_visualization.rb [hunk 1]
```diff
@@ -73,16 +73,20 @@ def add_circle_trace(node_id, _sampling_rate, stats)
...
+      ratio = 1 - distance_ratio
 
       # Transform circle definition to match current data sample
+      # dots shouldn't be scaled down below half the original size
+      scale_factor = Geometry.clamp(ratio, 0.5, 1.0)
       scaling = Geom::Transformation.scaling(scale_factor, 1.0, scale_factor)
       translation = Geom::Transformation.translation(position)
       transformation = translation * scaling
 
+      color_weight = ratio * 50.0
```
Comment (XPerianer) [inline src/system_simulation/trace_visualization.rb:85]: Maybe add a constant here

File: src/system_simulation/trace_visualization.rb [hunk 2]
```diff
@@ -73,16 +73,20 @@ def add_circle_trace(node_id, _sampling_rate, stats)
...
+
       @group = Sketchup.active_model.entities.add_group if @group.deleted?
       circle_instance = @group.entities.add_instance(circle_definition, transformation)
-      circle_instance.material = material_from_hsv(113, scale_factor * 130, 100)
+      circle_instance.material = material_from_hsv(117, 50 + color_weight, 100 - color_weight)
```
Comment (XPerianer) [inline src/system_simulation/trace_visualization.rb:89]: And also add constants here

File: src/ui/dialogs/spring_pane.rb [hunk 3]
```diff
@@ -207,7 +207,7 @@ def play_animation
   end
 
   def toggle_animation
-    simulate unless @simulation_data
+    simulate
```
Comment (XPerianer) [inline src/ui/dialogs/spring_pane.rb:210]: Is this related to the PR or left by accident?

Comment (birneamstiel) [reply src/ui/dialogs/spring_pane.rb:210]: Yep I wanted to include this fix here on purpose, but probably should have communicated better in the description :-)

File: src/system_simulation/trace_visualization.rb (hunk 1 above)
Comment (birneamstiel) [reply src/system_simulation/trace_visualization.rb:85]: Done in 898a351

File: src/system_simulation/trace_visualization.rb (hunk 2 above)
Comment (birneamstiel) [reply src/system_simulation/trace_visualization.rb:89]: Done in 898a351

**Comment by birneamstiel** [comment]
@XPerianer: Since @robertkovax wants to do some videos soon I'll go ahead and merge this, even if I made some minor changes you didn't review yet. If you like you could have a look post-merge at [this commit](https://github.com/robertkovax/TrussFab/pull/240/commits/0180920af94c3c6b7c6b826527dd075ecf6d91e2), that's the only one with some relevant changes you didn't check yet...
